### PR TITLE
Change multiple defaults

### LINKED ISSUE #75: Move mtry defaults from findBestSplit to DT and RF
Currently mtry is passed as a parameter to the `DecisionTree` or `RandomForest` constructor, then passed on to `findBestSplit`. The default should differ for DT and RF: `mtry` equal to the number of predictors and `mtry` equal to the square root of that number, respectively.

As such, move setting of `mtry` default to the parent tree object, and pass that on.

### LINKED ISSUE #76: Tree constructors to default to binary classification
Set `DecisionTree` and `RandomForest` constructor parameters to default to binary classification. Using Gini?

### PR DESCRIPTION
**Pull request by wfseaton**
- Move mtry defaults to tree level (DecisionTree = num_features_, RandomForest = sqrt(num_features)
- Set regression default as classification with loss gini

### CODE REVIEW

File: src/decision_tree.cpp [hunk 1]
```diff
@@ -260,37 +260,30 @@ std::pair<int,double> DecisionTree::findBestSplit(TreeNode *node) const
...
-        m = this->num_features_;
-        // Add indices without actually shuffling:
-        for (int i = 0; i < this->num_features_; i++){
-            shuf_inds[i] = i;
-        }
+    std::vector<int> shuf_inds(this->num_features_);
+    // Shuffle if mtry_ < num_features_ else deterministic
+    if (this->mtry_ < this->num_features_) {
+      // Create vector of column indices, equivalent to np.arange(0, df.shape[-1])
+      std::generate(shuf_inds.begin(), shuf_inds.end(), [n = 0] () mutable { return n++; });
```
Comment (johannes-kk) [inline src/decision_tree.cpp]: Could actually put this outside the `if` and do nothing with it unless the `if`-condition is met (which then only shuffles it).

**Review by johannes-kk — APPROVED** [review]
Made on comment, but nit-picking efficiency. LGTM!

File: src/decision_tree.cpp (hunk 1 above)
Comment (wfseaton) [reply src/decision_tree.cpp]: Great, updated

**Review by johannes-kk — APPROVED** [review]
Looks great!

**Pull request merged**